src/qz/communication/SerialProperties.java

```diff
@@ -72,4 +82,8 @@ public int getFlowControl() {
         return flowControl;
     }
 
+    public boolean isFile() {
+        return type == SerialDataType.FILE;
+    }
+
```

Comment:
Everything up to this point seems geared towards allowing more than just the two types, which is fine, but it means this has to be a `getType` to avoid having to refactor later if a new type is introduced. 
If we aren't expecting to add more types though, it would probably be better to refactor the enum & type variable out into a boolean "`fromFile`" value to return here.